### PR TITLE
lcd_touch: add new component CST816S

### PR DESCRIPTION
**Pull request by Lzw655**
# Checklist for new Board Support package or Component

- [x] Component contains License
- [x] Component contains README.md
- [x] Project [README.md](../README.md) updated
- [x] Component contains idf_component.yml file with `url` field defined
- [x] Component was added to CI [upload job](https://github.com/espressif/esp-bsp/blob/master/.github/workflows/upload_component.yml#L17)
- [ ] New files were added to CI build job
- [ ] _Optional:_ Component contains unit tests
- [ ] CI passing

# Change description
I have adapted a new touch chip for customers and want to store it in the components manager to provide convenience for other customers.

### CODE REVIEW

File: components/lcd_touch/esp_lcd_touch_cst816s/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+idf_component_register(SRCS "esp_lcd_touch_cst816s.c" INCLUDE_DIRS "include" REQUIRES "esp_lcd" "esp_lcd_touch")
```
[the remark this answers is not in the document]
Comment (Lzw655) [reply components/lcd_touch/esp_lcd_touch_cst816s/CMakeLists.txt]: Yes, you're right. I delete it [here](https://github.com/espressif/esp-bsp/pull/123/commits/32315fc350eaed34ba20105253df8b793e725511#diff-35665d04d6a96652fea2b153abf8995f736355c6960c7424f12cdb906f1647c3L1-R1) and also fix a CI error.

BTW, shall we also delete "esp_lcd_touch" in REQUIRES for other components in this PR? like [here](https://github.com/espressif/esp-bsp/blob/master/components/lcd_touch/esp_lcd_touch_ft5x06/CMakeLists.txt#L1).

File: components/lcd_touch/esp_lcd_touch_cst816s/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Get one X and Y coordinates with strength of touch.
+
+```
+    uint16_t touch_x[1];
+    uint16_t touch_y[1];
+    uint16_t touch_strength[1];
+    uint8_t touch_cnt = 0;
+
+    bool touchpad_pressed = esp_lcd_touch_get_coordinates(tp, touch_x, touch_y, &touch_strength, &touch_cnt, 1);
```
Comment (espzav) [inline components/lcd_touch/esp_lcd_touch_cst816s/README.md:62]:
```suggestion
    bool touchpad_pressed = esp_lcd_touch_get_coordinates(tp, touch_x, touch_y, touch_strength, &touch_cnt, 1);
```

**Review by espzav — APPROVED** [review]
@Lzw655 Thank you! LGTM, only one think in readme. 

Note: I cannot test it on HW, I haven't got this touch controller.

**Comment by Lzw655** [comment]
> @Lzw655 Thank you! LGTM, only one think in readme.
> 
> Note: I cannot test it on HW, I haven't got this touch controller.

Thank you for reviewing! Accepted it! And I also fix the same problems for other components, PTAL.

**Review by espzav — APPROVED** [review]
@Lzw655 Thank you very much for fix all issues! LGTM

**Comment by tore-espressif** [comment]
Unfortunately, all touch drivers have a public dependency on the base component `esp_lcd_touch`. This dependency is not defined anywhere now (nor in CMake, nor in idf_component.yml).

Our CI did not catch this problem. When we build BSP examples, we compile with BSP in esp-bsp repository, but other components (such us touch drivers) are fetched from component registry -> this should be improved.

I'll delete all touch component versions published from this PR and we will have to fix it.

cc @Lzw655 @igrr @espzav

**Comment by tore-espressif** [comment]
Fix here https://github.com/espressif/esp-bsp/pull/128 

Thanks @espzav

**Pull request merged**